aldryn_sites/utils.py

```diff
@@ -19,18 +19,21 @@ def set_site_names(force=False):
     _has_set_site_names = True
     sites = {site.id: site for site in Site.objects.all()}
     for site_id, site_config in settings.ALDRYN_SITES_DOMAINS.items():
+        config_name = site_config.get('name', site_config['domain'])
+        config_domain = site_config['domain']
+
         if site_id not in sites.keys():
             sites[site_id] = Site.objects.create(
                 id=site_id,
-                name=site_config['domain'],
-                domain=site_config['domain'],
+                name=config_name or config_domain,
+                domain=config_domain,
             )
-        site = sites[site_id]
-        if not site.name == site_config['domain']:
-            if site.name == site.domain:
-                # currently name and domain are the same... so change both to the new value
-                site.name = site_config['domain']
-            site.domain = site_config['domain']
+        else:
+            site = sites[site_id]
+            if not site.domain == config_domain:
+                if not config_name and site.name in (site.domain, 'example.com'):
+                    site.name = config_name
+                site.domain = config_domain
```

Comment:
if ``config_name`` is explicitly set (``name`` in ``ALDRYN_SITES_DOMAINS``), shouldn't it always be set?

in the ``else:`` part:
```
site = sites[site_id]
if not site.domain == config_domain:
    # domain needs to be updated
    if not config_name and site.name in (site.domain, 'example.com'):
        # ``name`` is not explicitly configured so we need to do something smart.
        # ``site.name`` is currently the same as ``site.domain`` in the database, 
        # so both should remain the same with the name being the ``domain``. 
        # Same if we have a freshly initialised database with the default django 
        # ``example.com`` entry.
        site.name = config_name
    site.domain = config_domain
if config_name:
    # a site ``name`` has explicitly been set. So update it.
    site.name = config_name
```

Ideally we'd also override the site admin to show the fields that have been overridden here as read only. But only for those ``site.id`` where that is actually the case. And ``name`` only if it was explicitly configured.